src/App.tsx

```diff
@@ -1,16 +1,63 @@
-import React from 'react';
+import React, { useCallback, useMemo, useState } from 'react';
+import debounce from 'lodash.debounce';
 import './App.scss';
 import { peopleFromServer } from './data/people';
+import { Dropdown } from './components/Dropdown';
+import { Person } from './types/Person';
 
-export const App: React.FC = () => {
-  const { name, born, died } = peopleFromServer[0];
+interface Props {
+  debounceDelay?: number;
+}
+
+export const App: React.FC<Props> = ({ debounceDelay = 300 }) => {
+  const [query, setQuery] = useState('');
+  const [focused, setFocused] = useState(false);
+  const [selectedPerson, setSelectedPerson] = useState<Person | null>(null);
+
+  const applyFilter = useMemo(
+    () =>
+      debounce((currentQuery: string) => {
+        setQuery(currentQuery);
```

Comment:
The debounce function is applied to `setQuery`, which might not be necessary. Consider applying debouncing to the filtering logic directly or ensuring that the query is only updated when necessary.